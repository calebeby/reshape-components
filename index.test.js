import components from '.'
import path from 'path'
import fs from 'mz/fs'
import reshape from 'reshape'
import expressions from 'reshape-expressions'

const fixture = name => async () => {
  const root = path.join(__dirname, 'fixtures', name)
  const filename = path.join(root, 'input.html')
  const input = (await fs.readFile(filename)).toString()
  const output = await reshape({
    plugins: [components({ root }), expressions()]
  })
    .process(input)
    .then(result => result.output())
  expect(output).toMatchSnapshot()
}

test('uses component from other file', fixture('component'))
