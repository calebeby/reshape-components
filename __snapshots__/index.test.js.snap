// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uses component from other file 1`] = `
"<import src=\\"./foobar.html\\" as=\\"FooBar\\"></import>
<div class=\\"g-foobar\\">
  <h1>This is a foobar called this is the title</h1>
  
  <p>Child Here</p>

  <h2>otherProp is 5</h2>
</div>

"
`;
